src/luger.erl

```diff
@@ -144,14 +145,46 @@ init_sink(State = #state{}, #syslog_udp_config{ host = Host,
                 syslog_udp_facility = Facility}.
 
 log(Priority, Channel, Message) ->
-    case whereis(?PROC_NAME) of
-        undefined -> {error, luger_not_running};
-        _ -> do_log(Priority, Channel, Message)
-    end.
+    log(Priority, Channel, Message, whereis(?PROC_NAME)).
 
-do_log(Priority, Channel, Message) ->
+log(_Priority, _Channel, _Message, undefined) ->
+    {error, luger_not_running};
+log(Priority, Channel, Message, _) ->
     [{state, State}] = ets:lookup(?TABLE, state),
 
+    Now = erlang:convert_time_unit(erlang:monotonic_time(), native, seconds),
+    Throttle = throttle_channel(Channel, Now, ets:lookup(?TABLE, Channel)),
+
+    log(State, Priority, Channel, Message, Throttle).
+
+
+log(_State, _Priority, _Channel, _Message, throttled) ->
+    throttled;
+log(State, Priority, Channel, Message, _) ->
+    record_metrics(State, Priority, Channel),
+    do_log(State, Priority, Channel, Message).
+
+
+throttle_channel(Channel, Now, [{Channel, _, LastUpdate}])
+  when Now > LastUpdate ->
+    ets:insert(?TABLE, {Channel, 1, Now});
+throttle_channel(Channel, Now, [{Channel, Freq, _}])
+  when Freq < ?THROTTLE_THRESHOLD ->
+    ets:insert(?TABLE, {Channel, Freq + 1, Now});
```

Comment:
Consider using `ets:update_counter/4` instead here.  I need to read that code path in the VM but I think it can be faster for this case; the update is destructive, so there'd be less consing.